File: hsqldb/java/read/hsqldb_read.java
// ---------------------------------------------------------------------
/*
	hsqldb/java/read/hsqldb_read.java

					Jun/12/2015

*/
// ---------------------------------------------------------------------
import java.sql.SQLException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.Statement;

// ---------------------------------------------------------------------
public class hsqldb_read
{
// ---------------------------------------------------------------------
public static void main (String args [])
	throws SQLException, ClassNotFoundException,
	InstantiationException, IllegalAccessException
{
	System.out.println ("*** 開始 ***");

	Class.forName ("org.hsqldb.jdbcDriver");

	String db_file = args[0];
	System.out.println (db_file);
	String database = "file:" + db_file + ";shutdown=true";

	Connection conn =
		DriverManager.getConnection("jdbc:hsqldb:" + database, "SA","");

	Statement ss = conn.createStatement ();

	rdb_common.display_proc (conn);

	ss.close ();

	conn.close();

	System.out.println ("*** 終了 ***");
}

// ---------------------------------------------------------------------
}

// ---------------------------------------------------------------------

File: hsqldb/scala/read/hsqldb_read.scala
// -----------------------------------------------------------------
//	read/hsqldb_read.scala
//
//						Jul/27/2010
// -----------------------------------------------------------------
import scala.io.Source
import java.sql._

// -----------------------------------------------------------------
object hsqldb_read
{
// -----------------------------------------------------------------
def main (args: scala.Array[String])
{
	println	("*** 開始 ***")

	hsqldb_display_proc ()

	println	("*** 終了 ***")
}

// -----------------------------------------------------------------
def hsqldb_display_proc ()
{
	val database = "file:/var/tmp/hsqldb/cities;shutdown=true"

	val str_connect= "jdbc:hsqldb:" + database

	Class.forName ("org.hsqldb.jdbcDriver")
	val conn = DriverManager.getConnection (str_connect,"SA","")

	sql_manipulate.display_proc (conn)

	conn.close ()
}

// -----------------------------------------------------------------
}
// -----------------------------------------------------------------
